firestore.rules

```diff
@@ -34,12 +34,17 @@ service cloud.firestore {
         function inPair(){
           return request.auth.uid in roomData().pair 
         }
+        function isStalker(){
+          return get(/databases/$(database)/documents/lobbies/$(code)/privatePlayers/$(request.auth.uid)).data.stalker == true;
+        }
         function canRead(){
           return inPair() || request.auth.uid in roomData().viewers ;
         }
+
+
         // firestore didn't like me using the above function for chatroom
         // I think its because the info is in the resource object
-        allow read: if request.auth.uid in resource.data.pair || request.auth.uid in resource.data.viewers ;
+        allow read: if request.auth.uid in resource.data.pair || request.auth.uid in resource.data.viewers || isStalker() ;
```

Comment:
`request.auth.uid in resource.data.viewers` already allow stalker to read chat. It wouldn't make it to your isStalker function because it'd already be true.

Reply:
![image](https://user-images.githubusercontent.com/98478781/200048524-750d06fe-9315-4a8d-b945-cf936e32e951.png)
We haven't added Stalker to the viewers list at this point yet! We need the stalker to be able to see the list of all chats _before_ we add him to a viewer list of a specific chat. See image for what happens if we don't have this.